src/main/cljc/jarl/core.cljc

```diff
@@ -1,28 +1,85 @@
 (ns jarl.core
-  (:require [jarl.encoding.json :as json]
+  (:require [clojure.string :as string]
+            [clojure.tools.cli :as cli]
+            [jarl.logging :as logging]
+            [jarl.encoding.json :as json]
             [jarl.io :as jio]
             [jarl.parser :as parser]
+            [jarl.eval :as evaluator]
+            [jarl.formatting :as fmt]
             #?(:cljs [cljs.nodejs :as nodejs]))
   #?(:clj (:gen-class)))
 
 #?(:cljs (enable-console-print!))
 #?(:cljs (nodejs/enable-util-print!))
 
-(defn run-first-plan [data input]
-  (let [[name plan] (first (get data :plans))]
-    (println "running plan" name)
-    (let [result (plan data input)]
-      (println "Result-set:" result)
-      result)))
-
-(defn -main
-  "Parses and runs a plan"
-  ([]
-   (let [loc #?(:clj "src/test/resources/rego/simple/plan.json"
-                :cljs (jio/path-resolve "src/test/resources/rego/simple/plan.json"))]
-     (run-first-plan (parser/parse-json (jio/read-file loc)) {})))
-  ([ir-file input-json]
-   (let [input (json/read-str input-json)]
-     (run-first-plan (parser/parse-json (jio/read-file ir-file)) input))))
+(def cli-options
+  [["-i" "--input string" "set input file path"
+    :parse-fn #(json/read-str (slurp %))]
+   ["-d" "--data string" "set data file path"
+    :parse-fn #(json/read-str (slurp %))]
+   [nil "--strict-builtin-errors" "treat built-in function errors as fatal"]
+   ["-h" "--help"]
+   ["-v" "--verbose" "Print debug information to stdout"]])
+
+(defn usage [options-summary]
+  (->> ["Jarl is a tool for evaluating Rego policies compiled into the OPA IR format"
+        ""
+        "Usage: jarl <IR file> [<entrypoint>] [options]"
+        ""
+        "Options:"
+        options-summary]
+       (string/join \newline)))
+
+(defn error-msg [errors]
+  (->> (flatten ["Failure:"
+                 ""
+                 errors
+                 ""
+                 "Use -h option for usage"])
+       (string/join \newline)))
+
+(defn parse-args
+  [args]
+  (let [{:keys [options arguments errors summary]} (cli/parse-opts args cli-options)]
+    (cond
+      errors
+      {:exit-message (error-msg errors)}
+      (:help options)
+      {:exit-message (usage summary) :ok? true}
+      (not-empty arguments)
+      (let [ir (first arguments)
+            plan (second arguments)]
+        (if (jio/file? ir)
+          {:ir ir :plan plan :options options}
+          {:exit-message (error-msg [(fmt/sprintf "IR \"%s\" is not a file" ir)])}))
+      :else
+      {:exit-message (usage summary)})))
+
+(defn exit [status msg]
+  (println msg)
+  (System/exit status))
```

Comment:
There are quite a few things specific to Java here. That’s fine of course, but we might want to move those to a .clj context and not a .cljc one. I don’t think we’ll need to support cljs for the “lein run” scenario.

Reply:
Oh, nice, at least the CLI library works for both 😃 https://github.com/clojure/tools.cli#clojurescript-support